genie-web/src/test/groovy/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobTerminationServiceImplSpec.groovy

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.rpc.grpc.services.impl.v4
+
+import com.netflix.genie.common.dto.JobStatus
+import com.netflix.genie.common.exceptions.GenieException
+import com.netflix.genie.common.exceptions.GenieServerException
+import com.netflix.genie.proto.*
+import com.netflix.genie.web.services.JobSearchService
+import io.grpc.stub.StreamObserver
+import spock.lang.Specification
+
+
+/**
+ * Specifications for the {@link GRpcJobTerminationServiceImpl} class.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+class GRpcJobTerminationServiceImplSpec extends Specification {
+
+    GRpcJobTerminationServiceImpl service
+    String jobId
+    JobKillListenerRequest request
+    JobKillListenerResponse response
+    JobSearchService jobSearchService = Mock()
+    StreamObserver<SyncResponse> responseObserver = Mock()
+
+    void setup() {
+        jobId = UUID.randomUUID().toString()
+        request = JobKillListenerRequest.newBuilder().setJobId(jobId).build()
+        response = JobKillListenerResponse.newBuilder().build()
+        service = new GRpcJobTerminationServiceImpl(jobSearchService)
+    }
+
+    def "Can kill unfinished jobs and clean up response observer"() {
+
+        service.registerForKillNotification(request, responseObserver)
```

Comment:
This is a call to the class under test. It should be in a `when` block as general rule/convention.